# Transfer Function and Superposition Theorem

### Chapter 2.6 - *DC Analysis*

### Example 2.7.2
A two-loop circuit is presented in Fig. 2.7.4.  

![Fig-2.7.4](Figures/Fig-2.7.4.png)

To find the transfer function that relates $J$ as the input to $U_3^J$ as the output, we should first find a transfer function ($K_{JI_3}$) that relates $J$ to $I_3$, the current flowing throught $R_3$.  And, of course, via Ohm's Law, $U_3^J=R_3I_3 = R_3K_{JI_3}J$.

$$ I_3 = K_{JI_3} J$$

Via Ohm's Law,  

$$ I_3 = \frac{U_3}{R_3}$$

Via KCL and Ohm's Law,  

$$ J = \frac{U_1}{R_1} + \frac{U_3}{R_3}$$

Since $E$ is short-circuited, $U_1=U_3$, thus,  

$$ J = \frac{U_3}{R_1} + \frac{U_3}{R_3} = U_3\,\frac{R_1+R_3}{R_1R_3}$$

$$ K_{JI_3} = \frac{I_3}{J} = \frac{R_1}{R_1+R_3}$$

Via the above and Ohm's Law,  

$$ U_3^J = \frac{R_3R_1}{R_1+R_3}\,J$$

To find $K_{EU_3}$:  

$$ U_3^E = K_{EU_3}E$$

Via KVL and Ohm's Law,  

$$ E = I(R_1+R_3)$$

and
$$ U_3^E=IR_3$$

Thus,  
$$ K_{EU_3} = \frac{U_3^E}{E} = \frac{R_3}{R_1+R_3}$$

## Drill Problems

#### Problem 1
1. The emf has increased its value two times (up to $2E$). Calculate the increment (value and
    sign) of the current $I$. Then, calculate the increment of the current source necessary to
    compensate this change.  
    
![Fig-P.2.7.1](Figures/Fig-P.2.7.1.png)

The transfer function ($K_{JI}$) that relates the input current source or excitation ($J=10\mathrm{A}$) to the output current ($I^J$) is given by:

\begin{equation}\tag{1.1}
    I^J = K_{JI}J = K_{JI}10\mathrm{A}
\end{equation}    

Via KCL and Ohm's Law,  

\begin{equation}\tag{1.2}
    I = 0.5\mathrm{S}\,U + 0.5\mathrm{S}\,U = 1\mathrm{S}\,U
\end{equation}    

and,
$$ 10\mathrm{A} - I - 1\mathrm{S}\,U = 0 $$

\begin{equation}\tag{1.3}
    J=10\mathrm{A} = I + 1\mathrm{S}\,U = 2\mathrm{S}\,U
\end{equation}    

\begin{equation}\tag{1.4}
    K_{JI} = \frac{I}{J} = \frac{1\mathrm{S}\,U}{2\mathrm{S}\,U} = 0.5
\end{equation}    

Now we find $K_{EI}$ such that:  

$$ I^E = K_{EI}E$$

Via Ohm's Law,  

$$ U_\mathrm{r} = 1\Omega\,I^E$$

Via KVL and Ohm's Law,  

$$ U_\mathrm{m} = - U_\mathrm{r}$$

$$ U_\mathrm{m} = 0.5E = -U_\mathrm{r} = -1\Omega\,I^E$$

$$ I^E = -0.5\mathrm{S}\,E$$

Thus,  

$$ K_{EI} = -0.5\mathrm{S}$$

$$ \Delta I^E = K_{EI}\Delta E = K_\mathrm{EI}E = -0.5\mathrm{S}\,10\mathrm{V} = -5\mathrm{A}$$

We need to find a $\Delta J$ such that:  

$$ \Delta I = \Delta I^E + \Delta I^J = 0$$

$$ \Delta I^E = -5\mathrm{A}$$

$$\Delta I^J = K_{JI}\Delta J = 5\mathrm{A}$$ 

$$ \Delta J = \frac{5\mathrm{A}}{K_{JI}} = 10\mathrm{A}$$

#### Problem 2
2. All independent sources (circuit inputs) and a shorted branch with the current $I_x$ (circuit output) are extracted from a linear circuit, as shown. With sources $J$ and $E_1$ on and $E_2=0$
    : $I_x = 20\mathrm{A}$ , with $J$ and $E_2$ on and $E_1=0$: $I_x=-5\mathrm{A}$ , with all three sources on: $I_x=12\mathrm{A}$. Find $I_x$ if, a) $J$ is doubled, b) $E_2$ is reversed.  
    
![Fig-P.2.7.3](Figures/Fig-P.2.7.3.png)

\begin{equation}\tag{2.1}
    I_x = 20\mathrm{A} = I_x^J + I_x^{E_1}
\end{equation}

\begin{equation}\tag{2.2}
    I_x = -5\mathrm{A} = I_x^J + I_x^{E_2}
\end{equation}

\begin{equation}\tag{2.3}
    I_x=12\mathrm{A}=I_x^J + I_x^{E_1} + I_x^{E_2}
\end{equation}

Via equations (2.3) and (2.2),  

\begin{equation}\tag{2.4}
    -17\mathrm{A} = I_x^{E_1}
\end{equation}

Via equations (2.3) and (2.1),  

\begin{equation}\tag{2.5}
    8\mathrm{A} = I_x^{E_2}
\end{equation}

Via equations (2.5), (2.4), and (2.3),  

\begin{equation}\tag{2.6}
    21\mathrm{A} = I_x^J
\end{equation}

Thus, 
\begin{equation}\tag{2.7}
    I_x = I_x^J + I_x^{E_2} + I_x^{E_1} = 12\mathrm{A}
\end{equation}    

To find the shorted branch's current *after* $J$ is doubled, denoted as '$I_x^*$':

$$ I_x^J = K{JI_x}J = \frac{G_1+G_2}{G_1+G_2+G_3}\,J = 21\mathrm{A}$$

$$ K_{JI_x} = \frac{I_x^J}{J} = \frac{21\mathrm{A}}{J}$$

$$ \Delta I_x^J = K_{JI_x} \Delta J = K_{JI_x}(2J - J) = K_{JI_x}J = 21\mathrm{A}$$

And,
$$ I_x^* = I_x + \Delta I_x^J = 33\mathrm{A}$$

To find $I_x$ *after* $E_2$ is reversed, denoted as '$I_x^*$':

$$ \Delta E_2 = -E_2 - E_2 = -2E_2$$

$$ I_x^{E_2} = K_{E_2 I_x} E_2$$

$$ K_{E_2 I_x} = \frac{I_x^{E_2}}{E_2} = \frac{8\mathrm{A}}{E_2}$$

$$ \Delta I_x^{E_2} = K_{E_2 I_x} \Delta E_2 = -2I_x^{E_2} = -16\mathrm{A}$$

Finally,
$$ I_x^* = I_x + \Delta I_x^{E_2} = -4\mathrm{A}$$

*Note that the rest of the information below is not needed to solve the problem.*

Via KCL and Ohm's Law,  

\begin{equation}\tag{2.7a}
    I_x^{E_1} = (G_1+G_2)U_x^{E_1}= -17\mathrm{A}
\end{equation}

and,
$$ I_x^{E_2} = (G_1 + G_2)U_x^{E_2}=8\mathrm{A}$$

Via KVL,
$$ U_J^{E_1} + U_x^{E_1} + E_1 = 0 $$

Via Ohm's $U_J^{E_1} = R_3I_x^{E_1} = -17\mathrm{A}\,R_3$

\begin{equation}\tag{2.7b}
    U_x^{E_1} = 17\mathrm{A}\,R_3 - E_1
\end{equation}    

Via equations (2.7a) and (2.7b),    

$$ (G_1 + G_2)\big(I_x^{E_1}R_3 + E_1\big) = -I_x^{E_1}$$

$$ I_x^{E_1}\big((G_1+G_2)R_3 + 1\big) = -E_1(G_1+G_2)$$

\begin{equation}\tag{2.7}
    I_x^{E_1} = K_{E_1 I_x}E_1 \quad \text{where}\; K_{E_1 I_x} = -\frac{G_1+G_2}{(G_1+G_2)R_3 + 1}    
\end{equation}  

#### Problem 4
4. Use the superposition theorem to find the voltage in Problems 2.3.5 and 2.3.6.

![Fig-P.2.3.5](Figures/Fig-P.2.3.5.png)

$$ I^{1\mathrm{A}} = K_{1\mathrm{A}\,I}1\mathrm{A}$$

Let $R_1:=4\Omega$ and $R_2:=12\Omega + 8\Omega = 20\Omega$.  It then follows that $I^{1\mathrm{A}}=U^{1\mathrm{A}}G_1=U^{1\mathrm{A}}0.25\mathrm{S}$.

$$ K_{1\mathrm{A}I} = \frac{G_1}{G_1+G_2} = \frac{0.25\mathrm{S}}{0.3\mathrm{S}}\approxeq 0.833$$

In [1]:
4*0.25/0.3

3.3333333333333335

Thus, 
$$ I^{1\mathrm{A}}\approxeq 833.33\mathrm{mA}$$

Via Ohm's Law,  

\begin{equation}\tag{4.1}
    U^{1\mathrm{A}} = 4\Omega\,I^{1\mathrm{A}} \approxeq 3.33\mathrm{V}
\end{equation}    

In [2]:
1/0.3

3.3333333333333335

Another way of finding the value of the same quantity is the following:  

Via KCL and Ohm's Law,  

$$ 1\mathrm{A} = G_1U^{1\mathrm{A}} + G_2U^{1\mathrm{A}} $$

$$ U^{1\mathrm{A}} = \frac{1\mathrm{A}}{G_1+G_2} \approxeq 3.33\mathrm{V} $$

In [3]:
1/((1/4)+(1/20))

3.3333333333333335

To find the response of $U$ to $E$, denoted as '$U^E$':  

Via KVL and Ohm's Law,  

$$ 20\Omega\,I - 6\mathrm{V} + 4\Omega I = 24\Omega\,I - 6\mathrm{V} = 0$$

$$ I = \frac{6\mathrm{V}}{24\Omega} = 250\mathrm{mA} $$

In [4]:
6/24

0.25

Via Ohm's Law,  

\begin{equation}\tag{4.2}
    U^E = -4\Omega\,I = -1\mathrm{V}
\end{equation}    

Another way of finding the value of the same quantity is to find the transfer function.  

$$ U^E = K_{EU} E$$

$$ K_{EU} = -\frac{4\Omega}{24\Omega} = -\frac{1}{6}$$

As for $U^J$ where $J=0.5\mathrm{A}$.

In [5]:
I_J = 0.5*(1/16)/((1/16)+(1/8))
U_J = 4*I_J
U_J

0.6666666666666666

\begin{equation}\tag{4.3}
    U^J = \frac{2}{3}\,\mathrm{V}
\end{equation}    

Another way of finding the value of same quantity:  

Via KCL,  

$$ 0.5\mathrm{A} = I_\mathrm{l} + I_\mathrm{r}$$

Via KVL and Ohm's Law,  

$$ 8\Omega\,I_\mathrm{l} = (12\Omega + 4\Omega)\,I_\mathrm{r}$$

$$ I_\mathrm{l} = 2I_\mathrm{r}$$

$$ 3I_\mathrm{r} = 0.5\mathrm{A}$$

$$ I_\mathrm{r} = \frac{1}{6}\,\mathrm{A}$$

$$ U^J = 4\Omega\,I_\mathrm{r} = \frac{2}{3}\,\mathrm{V}$$

$$ U = 3\mathrm{V} $$

#### Problem 5
5. All independent sources (circuit inputs) and a load branch with the current $I:=I_\mathrm{l}$ (circuit
    output) are extracted from a linear circuit, as shown. This current is measured for two
    different values of $E$ and $J$:
    1. $E=7\mathrm{V}$, $J=3\mathrm{A}$: $I_\mathrm{l} = 1\mathrm{A}$,
    2. $E=9\mathrm{V}$, $J=1\mathrm{A}$: $I_\mathrm{l} = 13\mathrm{A}$.  
    

Find:  
**a)** transfer functions $K_{EI}$ and $K_{JI}$;     
**b)** $I$ when $E=15\mathrm{V}$ and $J=9\mathrm{A}$;  
**c)** the increase in $J$ required to compensate for the increase of $E$ from $7\mathrm{V}$ to $9\mathrm{V}$.

\begin{equation}\tag{5.1}
    I = I^J + I^E \quad\quad I^J = K_{JI}J \quad\quad I^E = K_{EI}E
\end{equation}    

First, we formulate the above equations with the set of values found in entry (A.) of the problem.  

\begin{equation}\tag{5.1A}
    1\mathrm{A} = K_{JI}3\mathrm{A} + K_{EI}7\mathrm{V}
\end{equation}    

Next we formulate the equations in (5.1) with the values found in problem entry (B.).  

\begin{equation}\tag{5.1B}
    13\mathrm{A} = K_{JI}1\mathrm{A} + K_{EI}9\mathrm{V}
\end{equation}

Via multiplying throughout equation (5.1B) by $3$ and shuffling some terms around,  

\begin{equation}\tag{5.2}
    K_{JI}3\mathrm{A} = 39\mathrm{A} - K_{EI}27\mathrm{V}
\end{equation}    

Via equations (5.2) and (5.1A),  

$$ -38\mathrm{A} = -K_{EI}20\mathrm{V}$$

\begin{equation}\tag{5.3}
    K_{EI} = \frac{38\mathrm{A}}{20\mathrm{V}} = 1.9\mathrm{S}
\end{equation}    

Via equations (5.3) and (5.1B),  

$$K_{JI}1\mathrm{A} = 13\mathrm{A} - 1.9\mathrm{S}\,9\mathrm{V}$$

In [6]:
13-1.9*9

-4.099999999999998

\begin{equation}\tag{5.4}
    K_{JI} = -4.1
\end{equation}    

**b)**  

$$I = K_{EI}15\mathrm{V} + K_{JI}9\mathrm{A} = -8.4\mathrm{A}$$

In [7]:
1.9*15 - 4.1*9

-8.399999999999999

**c)**  We must find $\Delta J$ such that:  

$$ \Delta I = \Delta I^J + \Delta I^E = 0$$

Here,  
$$ \Delta I^E = K_{EI}\Delta E = 1.9\mathrm{S}\,2\mathrm{V} = 3.8\mathrm{A}$$

In [8]:
1.9*2

3.8

Thus,
$$ \Delta I^J = K_{JI}\Delta J= -3.8\mathrm{A}$$

$$ \Delta J = \frac{-3.8\mathrm{A}}{-4.1} \approxeq 926.83\mathrm{mA} $$

In [9]:
3.8/4.1

0.926829268292683

#### Problem 6
6. The circuit shown is driven by two independent sources. Find transfer functions in the
    linear relationship:  $U=K_{EU}E + K_{JU}J$  
    
![Fig-P.2.7.6](Figures/Fig-P.2.7.6.png)

$$ K_{EU} = \frac{7}{11}$$

$$ G_1 = \frac{1}{6\Omega}\quad \text{and} \quad G_2 = \frac{1}{16\Omega}$$

$$ -I^J = K_{JI}\,2\mathrm{A}$$

$$ K_{JI} = \frac{G_1}{G_1+G_2}\approxeq 0.73$$

In [10]:
G_1 = 1/6; G_2=1/16

In [12]:
1/(G_1+G_2)

4.363636363636364

$$ U^J = 6\Omega\,I^J = -\frac{1}{G_1+G_2}\,2\mathrm{A}$$

$$ K_{JU} = -\frac{1}{G_1+G_2} \approxeq  -4.36\Omega$$